
The perl module build pulls in cyrus' local <assert.h> instead of the one
from the system, and the definitions therein conflict.

Temporarily skip the definitions when building perl, this resolves the
problem pending a proper fix from upstream.

See https://github.com/cyrusimap/cyrus-imapd/issues/4123

diff -wpruN '--exclude=*.orig' a~/lib/assert.h a/lib/assert.h
--- a~/lib/assert.h	1970-01-01 00:00:00
+++ a/lib/assert.h	1970-01-01 00:00:00
@@ -43,6 +43,8 @@
 #ifndef INCLUDED_ASSERT_H
 #define INCLUDED_ASSERT_H
 
+#ifndef PERL_POLLUTE
+
 #ifdef __STDC__
 #define assert(ex)      {if (!(ex))assertionfailed(__FILE__, __LINE__, #ex);}
 void assertionfailed(const char *file, int line, const char *expr);
@@ -50,4 +52,6 @@ void assertionfailed(const char *file, i
 #define assert(ex)      {if (!(ex))assertionfailed(__FILE__, __LINE__, (char*)0);}
 #endif
 
+#endif
+
 #endif /* INCLUDED_ASSERT_H */
